src/topography.f90

```diff
@@ -464,6 +465,45 @@ subroutine topography_min_max_depth(this, vgrid_file, vgrid_type, level)
 
   end subroutine topography_min_max_depth
 
+ subroutine topography_cut_off_T_cells(this, hgrid, cutoff)
+    class(topography_t), intent(inout) :: this
+    character(len=*), intent(in) :: hgrid
+    real(real64), intent(in) :: cutoff
+
+    integer(int32) :: i, j
+    integer(int32) :: ncid_hgrid, dy_id          ! NetCDF ids for hgrid
+    integer(int32) :: dids_dy(2)                 ! NetCDF ids for dimensions
+    integer(int32) :: ny_len, nxp_len, nx_len    ! dimensions for hgrid
+    real(real64), allocatable :: dy(:,:)         ! To store dy variable from hgrid
+
+    ! Read hgrid to get dy
+    write(output_unit,'(3a)') "Attempting to open file '", trim(hgrid), "'"
+    call handle_error(nf90_open(trim(hgrid), nf90_nowrite, ncid_hgrid))
+    call handle_error(nf90_inq_varid(ncid_hgrid, 'dy', dy_id))
+    call handle_error(nf90_inquire_variable(ncid_hgrid, dy_id, dimids=dids_dy))
+    call handle_error(nf90_inquire_dimension(ncid_hgrid, dids_dy(1), len=ny_len))
+    call handle_error(nf90_inquire_dimension(ncid_hgrid, dids_dy(2), len=nxp_len))
+
+    ! Allocate memory for dy based on its dimensions
+    allocate(dy(ny_len, nxp_len))
```

Comment:
why is the index order different here from the rest of the code, which is x,y? e.g. here in `topography_constructor`
https://github.com/COSIMA/domain-tools/blob/ab469dfcdb038633f46f6f17ed84a2efb8aa46e1/src/topography.f90#L66

Reply:
@aekiss  `dy` is read from `ocean_hgrid.nc` and the index order of `dy` in the netcdf file is

```
	double dy(ny, nxp) ;
		dy:standard_name = "grid_edge_y_distance" ;
		dy:units = "meters" ;
```

Reply:
Hm, now I'm confused - that's the same index order as `depth` in `topog.nc`
```
$ ncdump -h topog.nc
netcdf topog {
dimensions:
	nx = 1440 ;
	ny = 1080 ;
variables:
	float depth(ny, nx) ;
```

Reply:
I think netCDF is row-major order, while Fortran is column-major order, no?

Reply:
Yes, NetCDF is row-major, and Fortran is column-major. But I think the NetCDF Fortran library automatically handles the conversion, so we don’t need to do it manually.

Reply:
No, the library does not transposes the data.

The code works because when you get the dimensions ID with `nf90_inquire_variable`  they come in column-major order (in this case `[nx_id, ny_id]`). Then what you call 'y' (`dids_dy(1)`) is actually 'x' and what you call 'x' (`dids_dy(2)`) is actually 'y'.

This is easy to check. Just print the values of `nx_len` and `ny_len`.

Reply:
Thanks @aekiss and @micaeljtoliveira, I have changed the indices in the allocation, similar to the rest of the code in this commit [a33cbb4](https://github.com/COSIMA/domain-tools/pull/41/commits/a33cbb4045a0bc834070266a3f1f9d6c5ca926a6)

I have also changed the loop, without changing the indices of `dy(2 * i - 1, 2 * j) + dy(2 * i, 2 * j)` , since I changed the allocation to `allocate(dy(nxp_len, ny_len))` and also swapped the i and j in do loop, I believe no need to change `dy(2 * i - 1, 2 * j) + dy(2 * i, 2 * j)`

```
-    do i = 1, ny_len / 2
-      do j = 1, (nxp_len - 1) / 2
+    do j = 1, ny_len / 2
+      do i = 1, (nxp_len - 1) / 2
           if (dy(2 * i - 1, 2 * j) + dy(2 * i, 2 * j) < cutoff) then  !Input cutoff in meters
             this%depth(i, j) = MISSING_VALUE  ! Set values below cutoff to zero or another value as needed
           end if
           ```

Reply:
Hi @ezhilsabareesh8 Thanks for the changes!

I think you also need to change the lines where you read `ny_len` and `nxp_len`.

Reply:
Thanks @micaeljtoliveira , this is how `ny_len` and `nxp_len` are read.

```
    call handle_error(nf90_inquire_variable(ncid_hgrid, dy_id, dimids=dids_dy))
    call handle_error(nf90_inquire_dimension(ncid_hgrid, dids_dy(1), len=ny_len))
    call handle_error(nf90_inquire_dimension(ncid_hgrid, dids_dy(2), len=nxp_len))
 ```
    
 Is it fine, since `dy(ny, nxp)`?

Reply:
I think it should be:
```fortran
    call handle_error(nf90_inquire_variable(ncid_hgrid, dy_id, dimids=dids_dy))
    call handle_error(nf90_inquire_dimension(ncid_hgrid, dids_dy(1), len=nxp_len))
    call handle_error(nf90_inquire_dimension(ncid_hgrid, dids_dy(2), len=ny_len))
```
Basically in the Fortran API `dids_dy(:)` is ordered in the same way as the dimensions of the corresponding Fortran array.

Reply:
Thanks @micaeljtoliveira. I have change it in this commit 2833aeb